mgmtsystem_action/models/mgmtsystem_action.py

```diff
@@ -0,0 +1,185 @@
+    active = fields.Boolean('Active', default=True)
+    opening_date = fields.Datetime('Opening Date', readonly=True)
+    create_date = fields.Datetime('Creation Date', readonly=True)
+    date_deadline = fields.Date('Deadline')
+    date_closed = fields.Datetime('Closed', readonly=True)
+    date = fields.Date('Action Date', default=fields.datetime.now(),
+                       select=True)
+    reference = fields.Char('Reference', required=True,
+                            readonly=True, default="NEW")
+    user_id = fields.Many2one(
+        'res.users', 'Responsible', default=owner)
+    company_id = fields.Many2one('res.company', 'Company')
+    description = fields.Text('Description')
+    type_action = fields.Selection(
+        [
+            ('immediate', 'Immediate Action'),
+            ('correction', 'Corrective Action'),
+            ('prevention', 'Preventive Action'),
+            ('improvement', 'Improvement Opportunity')
+        ], 'Response Type')
+
+    system_id = fields.Many2one('mgmtsystem.system', 'System')
+    company_id = fields.Many2one('res.company', 'Company', default=own_company)
+
+    stage_id = fields.Many2one(
+        'mgmtsystem.action.stage',
+        'Stage',
+        default=lambda self: self.get_default_stage()
+    )
+
+    @api.model
+    def get_default_stage(self):
+        """Return the default stage."""
+        return self.env['mgmtsystem.action.stage'].search([
+            ('is_starting', '=', True)
+        ]).id
+
+    @api.model
+    def create(self, vals):
+        """Creation of Action."""
+        vals.update({
+            'reference': self.env['ir.sequence'].next_by_code(
+                'mgmtsystem.action'
+            )
+        })
+        if vals.get('opening_date'):
+            vals['opening_date'] = None
+
+        template = self.env.ref(
+            'mgmtsystem_action.email_template_new_action_reminder')
+        action = super(MgmtSystemAction, self).create(vals)
+        template.send_mail(action.id, force_send=True)
+        return action
+
+    @api.multi
+    def write(self, vals):
+        """Update user data."""
+        stage_open = self.env.ref('mgmtsystem_action.stage_open')
+        stage_close = self.env.ref('mgmtsystem_action.stage_close')
+        if vals.get('stage_id'):
+            if vals['stage_id'] == stage_open.id:
+                vals['opening_date'] = datetime.now()
+            if vals['stage_id'] == stage_close.id:
+                vals['date_closed'] = datetime.now()
+
+        result = super(MgmtSystemAction, self).write(vals)
+        return result
+
+    def send_mail_for_action(self, action):
+        """Set a document state as draft and notified the reviewers."""
+        template = self.env.ref(
+            'mgmtsystem_action.email_template_new_action_reminder')
+
+        template.send_mail(action.id, force_send=True)
+        return True
+
+#    @api.multi
+#    def message_auto_subscribe(self, updated_fields, values=None):
```

Comment:
You can remove this now: this feature is automatically provided by the `mail` module.